packages/palette/src/elements/Modal/ModalBase.tsx

```diff
@@ -0,0 +1,167 @@
+import React, { useEffect, useRef, useState } from "react"
+import { createPortal } from "react-dom"
+import { RemoveScroll } from "react-remove-scroll"
+import styled from "styled-components"
+import { zIndex as systemZIndex, ZIndexProps } from "styled-system"
+import { useCursor } from "../../utils/useCursor"
+import { Flex, FlexProps } from "../Flex"
+
+// TODO: Update TypeScript definitions for this library
+// https://github.com/theKashey/react-remove-scroll
+const ScrollIsolation = styled(RemoveScroll as any)`
+  width: 100%;
+  height: 100%;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+`
+
+const Container = styled(Flex)`
+  position: fixed;
+  top: 0;
+  right: 0;
+  bottom: 0;
+  left: 0;
+  align-items: center;
+  justify-content: center;
+  ${systemZIndex}
+`
+
+const Dialog = styled(Flex).attrs({ role: "dialog" })`
+  overflow-y: auto;
+  -webkit-overflow-scrolling: touch;
+
+  > * {
+    max-height: 100%;
+  }
+`
+
+/**
+ * BaseModal
+ */
+export type ModalBaseProps = FlexProps &
+  ZIndexProps & {
+    children?: React.ReactNode
+    dialogProps?: FlexProps
+    onClose?(): void
+  }
+
+/**
+ * It seems we've landed on this value as the 'top'
+ */
+export const DEFAULT_MODAL_Z_INDEX = 9999
```

Comment:
Now that Reaction is inside of Force we can actually extract a comprehensive z-index map. I'll give that a shot as well.